### DISCUSSION TITLE
HTMLHelper::_('behavior.modal', 'a.modal'); equivalent

### DISCUSSION THREAD

**Discussion by joomdonation** [joomla/joomla-cms#35136] [Q&A] [answered]
In Joomla 3, we have `HTMLHelper::_('behavior.modal', 'a.modal');` to allow open a link on modal. Do we have equivalent code for Joomla 4 @dgrammatiko @Fedik ? Need it while converting weblinks extension to J4.

**Comment by dgrammatiko** [comment, marked as answer]
No, there's only the Bootstrap version in J4. The reason is that this method was based on Mootools and J4 dropped Mootools without a replacement. There was supposed to be a replacement with https://github.com/joomla-projects/custom-elements/tree/master/src/js/modal and maybe it's something that the project should pursue. I remember @Fedik was asking for a better (client-side programmable) modal and the custom elements could do that. Anyways for the moment, you have to convert it to Bootstrap... 🤷‍♂️

**Reply by joomdonation** [reply]
Thanks @dgrammatiko. Any any code example which I can use? Or maybe I can just remove the open on modal option for weblink item :D.

**Reply by dgrammatiko** [reply]
What's the repo/file link? I could provide some pseudocode

**Reply by joomdonation** [reply]
Great. Many thanks. Here is the link to the current code https://github.com/joomdonation/weblinks/blob/4.0-compatible/src/components/com_weblinks/tmpl/category/default_items.php#L127

**Reply by dgrammatiko** [reply]
So this:
```php
// Open in a modal window
HTMLHelper::_('behavior.modal', 'a.modal');
echo '<a class="modal" href="' . $link . '"  rel="{handler: \'iframe\', size: {x:' . $this->escape($width) . ', y:' . $this->escape($height) . '}}">' .
$this->escape($item->title) . ' </a>';
break;
```

could be written as:
```php
$randomId = base64_encode($item->title);
$modalParams['title']  = $this->escape($item->title);
$modalParams['url']    = $link;
$modalParams['height'] = '100%';
$modalParams['width']  = '100%';
$modalParams['bodyHeight'] = 70;
$modalParams['modalWidth'] = 80;
echo HTMLHelper::_('bootstrap.renderModal', 'modal-' . $randomId, $modalParams);

echo '<button type="button" class="btn btn-link" data-bs-toggle="modal" data-bs-target="#modal-' . $randomId . '">
  Some Translatable string (for A11y)
</button>';
break;
```

**Reply by joomdonation** [reply]
Great. Thanks for helping. Will try to use that code.

**Reply by joomdonation** [reply]
Just want to update that the code works. Thanks again @dgrammatiko.
